src/web/mjs/engine/DiscordPresence.mjs

```diff
@@ -0,0 +1,167 @@
+        // get chapter pages (Reader/Downloader)
+        document.addEventListener( EventListener.onChapterGetPages, this._onChapterGetPages.bind(this) );
+
+        this.startDiscordPresence();
+    }
+
+    _onSettingsChanged() {
+        switch (this._settings.discordPresence.value) {
+            case 'none':
+                this.enabled = false;
+                this.enabledHentai = false;
+                break;
+            case 'nohentai':
+                this.enabled = true;
+                this.enabledHentai = false;
+                break;
+            case 'hentai':
+                this.enabled = true;
+                this.enabledHentai = true;
+                break;
+            default:
+                this.enabled = false;
+                this.enabledHentai = false;
+        }
+
+        if (this.enabled) {
+            this.statusNew = true;
+            this.startDiscordPresence();
+        } else {
+            this.stopDiscordPresence();
+        }
+    }
+
+    _onSelectConnector(e) {
+        // Hentai check
+        if(e.detail.tags.includes('hentai') || e.detail.tags.includes('porn')) {
+            this.hentai = true;
+        } else {
+            this.hentai = false;
+        }
+
+        this.status['details'] = 'Browsing ' + e.detail.label;
+        if (this.status.state) delete this.status.state;
+        this.status.startTimestamp = + new Date();
+        this.statusNew = true;
+        if (this.enabled && !this.rpc) this.startDiscordPresence();
+    }
+
+    _onMangaClicked(e) {
+        // Hentai check
+        if(e.detail.connector.tags.includes('hentai') || e.detail.connector.tags.includes('porn')) {
+            this.hentai = true;
+        } else {
+            this.hentai = false;
+        }
+
+        this.status['details'] = 'Browsing ' + e.detail.connector.label;
+        this.status['state'] = 'Looking at ' + e.detail.title;
+        this.status.startTimestamp = + new Date();
+        this.statusNew = true;
+        if (this.enabled && !this.rpc) this.startDiscordPresence();
+    }
+
+    _onChapterGetPages(e) {
+        // Hentai check
+        if(e.detail.manga.connector.tags.includes('hentai') || e.detail.manga.connector.tags.includes('porn')) {
+            this.hentai = true;
+        } else {
+            this.hentai = false;
+        }
+
+        this.status['details'] = 'Reading ' + e.detail.manga.title;
+        this.status['state'] = e.detail.title;
+        this.status.startTimestamp = + new Date();
+        this.statusNew = true;
+        if (this.enabled && !this.rpc) this.startDiscordPresence();
+    }
+
+    async updateStatus() {
+        if(this.rpc) {
+            if (this.enabled && this.statusNew) {
+                this.IpcBytes = this.rpc.transport.socket.bytesWritten;
+                if( !this.hentai || (this.hentai && this.enabledHentai)) {
+                    this.rpc.setActivity(this.status);
```

Comment:
If supported, may set a custom type, because HakuNeko is not really a game that is played
https://discord.com/developers/docs/topics/gateway#activity-object-activity-structure

Reply:
Yeah i saw that. Unfortunately discord-rpc doesn't support it (again):
![image](https://user-images.githubusercontent.com/18627623/86149270-0e243400-bafc-11ea-9c13-fa096e9b9c00.png)